controller/konnect/konnectextension_controller_utils.go

```diff
@@ -33,39 +29,95 @@ import (
 	konnectv1alpha1 "github.com/kong/kubernetes-configuration/api/konnect/v1alpha1"
 )
 
-// getKonnectControlPlane retrieves the Konnect Control Plane based on the provided KonnectExtension specification.
+// getGatewayKonnectControlPlane retrieves the Konnect Control Plane form K8s cluster
+// based on the provided KonnectExtension specification.
 // It supports two types of ControlPlaneRef: KonnectNamespacedRef and KonnectID.
 //
 // Returns:
 // - cp: The retrieved Konnect Control Plane.
 // - res: The result of the controller reconciliation.
 // - err: An error if the retrieval fails.
-func (r *KonnectExtensionReconciler) getKonnectControlPlane(
+func (r *KonnectExtensionReconciler) getGatewayKonnectControlPlane(
 	ctx context.Context,
-	logger logr.Logger,
-	sdk sdkops.ControlPlaneSDK,
 	ext konnectv1alpha1.KonnectExtension,
 	dependingConditions ...metav1.Condition,
-) (cp *sdkkonnectcomp.ControlPlane, res ctrl.Result, err error) {
+) (cp *konnectv1alpha1.KonnectGatewayControlPlane, res ctrl.Result, err error) {
+	// Get respective KonnectGatewayControlPlane from K8s cluster.
+	var getFromK8sErr error
```

Comment:
According to https://go.dev/wiki/Errors#naming error variables should start with `err` or `Err`.